### ISSUE TITLE
Reload on dotenv hidden files

### ISSUE THREAD

**Issue by brauliobo** [alexch/rerun#117]
The default ignore pattern doesn't allow to reload on `.env` files of https://github.com/bkeepers/dotenv

**Comment by alexch** [comment]
Unfortunately, *all* files beginning with dot are ignored. This is a limitation of the file-change-monitoring library we're using -- I'm not sure if it's inside the Listen gem itself, or lower down inside `fsevent` or whatever. If you happen to research this, please let me know what you find. Learn more at <https://github.com/guard/listen#listen-adapters>

**Comment by brauliobo** [comment]
I've used before the `listen` gem to watch the `.env` files

**Comment by alexch** [comment]
Good point! Here is the line where we add all dotfiles to the ignore list:

https://github.com/alexch/rerun/blob/master/lib/rerun/watcher.rb#L86

I believe I did this in order to match the shell-glob behavior (where `*` matches everything *except* dotfiles). Why didn't I ever add a `--no-ignore-dotfiles` option? I have a dim memory that I couldn't get it to work consistently across different adapters, but that may have been fixed once I moved over to using the Listen backend. 

It shouldn't be difficult to implement; the hard(ish) part will be to write tests demonstrating the desired behvaior and testing it across different adapters/platforms.
